### PR TITLE
Add multi_gru_seq_fuse_pass and tests

### PR DESCRIPTION
**Pull request by wojtuss**
### PR types
New features

### PR changes
Others

### Describe
This PR introduces a `multi_gru_seq_fuse_pass` which fuses two subsequent `multi_gru` ops into a multilayer `multi_gru` op.
This fuse is the last step in transformation of a sequence of concatenated pairs of `fusion_gru` ops into one multilayer `multi_gru` op.

This PR is a part of the draft PR #28230.
It is related to the GRU INT8 model optimization (#27330).

### CODE REVIEW

File: paddle/fluid/framework/ir/mkldnn/multi_gru_seq_fuse_pass.cc [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+namespace paddle {
+namespace framework {
+namespace ir {
+
+using EigenVectorArrayMap = Eigen::Map<Eigen::Array<double, Eigen::Dynamic, 1>>;
+using string::PrettyLogDetail;
+
+namespace {
+
+void UnlinkNodes(ir::Node* a, ir::Node* b) {
```
Comment (wozna) [inline paddle/fluid/framework/ir/mkldnn/multi_gru_seq_fuse_pass.cc]: This function is not used anywhere in this file. Will you need it later or in another file?

File: paddle/fluid/framework/ir/mkldnn/multi_gru_seq_fuse_pass.cc [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+std::vector<std::string> join_inputs(Node* op1, Node* op2,
+                                     std::string input_name) {
+  auto in1 = op1->Op()->Input(input_name);
+  auto& in2 = op2->Op()->Input(input_name);
+  in1.insert(in1.end(), in2.begin(), in2.end());
+  return in1;
+}
+
+void link_inputs_by_name(ir::Graph* graph, std::vector<std::string> input_names,
```
Comment (wozna) [inline paddle/fluid/framework/ir/mkldnn/multi_gru_seq_fuse_pass.cc]: This function also is not used anywhere in this file. Please delete it if not needed.

File: paddle/fluid/framework/ir/mkldnn/multi_gru_seq_fuse_pass.cc (hunk 1 above)
Comment (wojtuss) [reply paddle/fluid/framework/ir/mkldnn/multi_gru_seq_fuse_pass.cc]: Done.

File: paddle/fluid/framework/ir/mkldnn/multi_gru_seq_fuse_pass.cc (hunk 2 above)
Comment (wojtuss) [reply paddle/fluid/framework/ir/mkldnn/multi_gru_seq_fuse_pass.cc]: Done.

**Review by wozna — APPROVED** [review]
Very clean code. Everything looks fine to me

**Comment by wojtuss** [comment]
@luotao1 , could you please add your review?

**Review by luotao1 — APPROVED** [review]

**Pull request merged**